xbmc/Application.cpp

```diff
@@ -767,26 +767,35 @@ bool CApplication::Initialize()
 
     std::vector<std::string> incompatibleAddons;
     event.Reset();
-    std::atomic<bool> isMigratingAddons(false);
-    CJobManager::GetInstance().Submit([&event, &incompatibleAddons, &isMigratingAddons]() {
-        incompatibleAddons = CAddonSystemSettings::GetInstance().MigrateAddons([&isMigratingAddons]() {
-          isMigratingAddons = true;
-        });
-        event.Set();
-      }, CJob::PRIORITY_DEDICATED);
-    localizedStr = g_localizeStrings.Get(24151);
-    iDots = 1;
-    while (!event.WaitMSec(1000))
+
+    // Addon migration
+    if (!CServiceBroker::GetAddonMgr().GetIncompatibleAddons().empty())
     {
-      if (isMigratingAddons)
-        CServiceBroker::GetRenderSystem()->ShowSplash(std::string(iDots, ' ') + localizedStr + std::string(iDots, '.'));
-      if (iDots == 3)
-        iDots = 1;
-      else
-        ++iDots;
+      CJobManager::GetInstance().Submit(
+          [&event, &incompatibleAddons]() {
+            if (CServiceBroker::GetRepositoryUpdater().CheckForUpdates())
```

Comment:
should this be guarded by checking whether auto updates are enabled?

Reply:
This doesn't update addons, only gets the list of addons from the installed repos. The update process for addons is guarded by the autoupdate setting